### PR TITLE
Add load balancer to allowed hosts for aws deployment

### PR DESCRIPTION
**Pull request by andrewpeng02**
# Add load balancer to allowed hosts for aws deployment

**What user problem are we solving?**
Because the `ALLOWED_HOSTS` is empty, the backend was refusing to return requests to the aws load balancer

**What solution does this PR provide?**
Adds the relevant ips to allowed hosts

**Testing Methodology**
Change redirect for /api/training to use `http://backend-load-balancer-296304048.us-east-1.elb.amazonaws.com` instead of localhost in `next.config.js` and try to train a new model

https://github.com/DSGT-DLP/Deep-Learning-Playground/assets/47485510/4194f682-d7f6-4088-9c15-1fdfbb08193b

### CODE REVIEW

File: training/training/middleware/health_check_middleware.py
Comment (andrewpeng02) [reply training/training/middleware/health_check_middleware.py]: Not sure why, should be fine
image

File: training/training/settings.py
Comment (karkir0003) [inline training/training/settings.py:31]: can u explain this part? can we ensure the dockerization will allow backend to be accessible from the container?

**Review by farisdurrani — APPROVED** [review]

**Comment by farisdurrani** [comment]
lgtm, ignore the githubactions errors if you're confident it's working as intended. Unfortunately GA does not have access to a working Python env to make accurate statements. Wait until tomorrow or until another PL approves before you merge

File: training/training/middleware/health_check_middleware.py
Comment (karkir0003) [reply training/training/middleware/health_check_middleware.py:11]: What's the type for `get_response`? @andrewpeng02

**Review by karkir0003 — COMMENTED** [review]
added some comments @andrewpeng02. please address

**Comment by karkir0003** [comment]
> lgtm, ignore the githubactions errors if you're confident it's working as intended. Unfortunately GA does not have access to a working Python env to make accurate statements. Wait until tomorrow or until another PL approves before you merge

I wonder if we can ensure that in our github actions checks? I can try digging into this for a bit, but not the highest priority problem to solve

File: training/training/settings.py
Comment (andrewpeng02) [reply training/training/settings.py:31]: The if statement is only necessary on ECS since the requests come from different ip addresses, just dockerizing it by itself doesn't need it

File: training/training/middleware/health_check_middleware.py
Comment (karkir0003) [reply training/training/middleware/health_check_middleware.py:11]: @andrewpeng02 I'm referring to the `get_response` parameter in the constructor of the `HealthCheckMiddleware` class

File: training/training/settings.py
Comment (karkir0003) [reply training/training/settings.py:31]: I see. Got it.

File: training/training/middleware/health_check_middleware.py
Comment (andrewpeng02) [reply training/training/middleware/health_check_middleware.py:11]: It's a callable, it's a required parameter for a middlware

Comment (karkir0003) [reply training/training/middleware/health_check_middleware.py:11]: @andrewpeng02 can you add the type hint?

**Pull request merged**
